### PR TITLE
Remove

### PR DESCRIPTION
**Pull request by rjwright**
This is a work in progress and it needs tests.

### CODE REVIEW

**Comment by rjwright** [comment]
@shans @dstockwell I think this is almost ready so feel free to have a look.

File: src/group-constructors.js [hunk 1]
```diff
@@ -185,7 +179,7 @@
       }
     };
 
-    underlyingAnimation = scope.timeline.play(new scope.KeyframeEffect(null, ticker, group._timing));
+    underlyingAnimation = scope.timeline._play(new scope.KeyframeEffect(null, ticker, group._timing));
```
Comment (shans) [inline src/group-constructors.js]: Why did you need to do this?

File: src/keyframe-effect-constructor.js [hunk 2]
```diff
@@ -14,6 +14,41 @@
...
+      if (effect._parent) {
+        if (oldParents.indexOf(effect._parent) == -1) {
+          oldParents.push(effect._parent);
+        }
+        effect._parent.children.splice(effect._parent.children.indexOf(effect), 1);
+        effect._parent = null;
+        disassociate(effect);
+      } else if (effect._animation && (effect._animation.effect == effect)) {
+        effect._animation.cancel();
+        effect._animation.effect = new KeyframeEffect(null, []);
```
Comment (shans) [inline src/keyframe-effect-constructor.js]: Maybe add a TODO stating that animation.effect should be allowed to be null?

File: src/timeline.js [hunk 3]
```diff
@@ -37,7 +37,7 @@
         return animation.playState != 'finished' && animation.playState != 'idle';
       });
     },
-    play: function(effect) {
+    _play: function(effect) {
```
Comment (shans) [inline src/timeline.js]: Oh. Never mind :)

**Comment by shans** [comment]
LGTM, merge once you've decided whether we should add that TODO or not.
